npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/vulnerability_auditor.rb

```diff
@@ -74,7 +75,11 @@ def audit(dependency:, security_advisories:)
               function: "npm:vulnerabilityAuditor",
               args: [Dir.pwd, vuln_versions]
             )
-            return fix_unavailable unless viable_audit_result?(audit_result, security_advisories)
+
+            validation_result = validate_audit_result(audit_result, security_advisories)
+            unless viable_audit_result?(validation_result)
+              return explain_fix_unavailable(validation_result, fix_unavailable)
```

Comment:
Subjective/non-blocking: if `explain_fix_unavailable` accepted `validation_result` as its only argument and simply returned the explanation as a string, it wouldn't need to know/restrict how the explanation was going to be used. This is just a helper method, so no big deal.

Reply:
Good suggestion. I went with that approach in https://github.com/dependabot/dependabot-core/pull/5542/commits/620491327d8a6e9bf9624ffec771108d53e16ee3